tesseract_common/CMakeLists.txt

```diff
@@ -12,13 +12,22 @@ endif()
 include(cmake/tesseract_macros.cmake)
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/")
 
-find_package(
-  Boost REQUIRED
-  COMPONENTS system
-             filesystem
-             serialization
-             stacktrace_noop
-  OPTIONAL_COMPONENTS stacktrace_backtrace stacktrace_basic)
+if(NOT APPLE)
+  find_package(
+    Boost REQUIRED
+    COMPONENTS system
+               filesystem
+               serialization
+               stacktrace_noop
+    OPTIONAL_COMPONENTS stacktrace_backtrace stacktrace_basic)
+else()
+  find_package(
+    Boost REQUIRED
+    COMPONENTS system
+               filesystem
+               serialization
+               stacktrace_noop)
+endif()
```

Comment:
Why does optional components cause an issue?

Reply:
`stacktrace_backtrace` was causing a linking error. We can create an issue to figure out the problem later.